packages/flutter_tools/lib/src/runner/local_engine.dart

```diff
@@ -165,26 +171,38 @@ class LocalEngineLocator {
     return 'host_$tmpBasename';
   }
 
-  EngineBuildPaths _findEngineBuildPath(String? localEngine, String enginePath) {
-    if (localEngine == null) {
-      throwToolExit(_userMessages.runnerLocalEngineRequired, exitCode: 2);
+  EngineBuildPaths _findEngineBuildPath(String? localEngine, String? localWebSdk, String enginePath) {
```

Comment:
Do we still need my recent changes above (in `_getHostEngineBasename()`)?

Reply:
Yes, I think it's best to keep that logic in there. If we point at a wasm out folder, we want to avoid all that special logic mutating the base name.